### PR TITLE
Sometimes err.url is undefined. so It must has protect code.

### PR DESCRIPTION
**Pull request by mixed**
When I ran `lighthouse-cli` from `https://www.inflearn.com/`. I found error. Sometimes err.url is undefined. so It must has protect code.
```
node lighthouse-cli --config-path=lighthouse-core/config/dobetterweb.json  https://www.inflearn.com/
```

### CODE REVIEW

**Comment by ebidel** [comment]
Good catch. It's nice to have URLs that expose issues.

Can you also add this for the other two audits that recently landed:
- https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/dobetterweb/no-mutation-events.js#L79
- https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/dobetterweb/uses-passive-event-listeners.js#L74

File: lighthouse-core/audits/dobetterweb/no-datenow.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ class NoDateNowAudit extends Audit {
     const pageHost = url.parse(artifacts.URL.finalUrl).host;
     // Filter usage from other hosts.
     const results = artifacts.DateNowUse.usage.filter(err => {
-      return url.parse(err.url).host === pageHost;
+      return url.parse(err.url || '').host === pageHost;
```
Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/no-datenow.js]: For stacktraces that are missing `.url`, what do you think about over communicating and reporting those cases as same-domain violation? In other words:

`return err.url ? url.parse(err.url).host === pageHost : true;`

Comment (mixed) [inline lighthouse-core/audits/dobetterweb/no-datenow.js]: @ebidel 
The displayed that `url` was `undefined` like below image.  What do you think about display? I think that `-` is more good than `undefined`.

image

Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/no-datenow.js]: Agreed. Line/col number without an URL makes them less useful, so let's keep what you have then. It looks like it finds all the violations on www.inflearn.com just fine :)

screen shot 2016-11-02 at 10 51 54 am

File: lighthouse-core/test/audits/dobetterweb/no-mutation-events-test.js [hunk 2]
```diff
@@ -48,4 +48,14 @@ describe('Page does not use mutation events', () => {
     assert.equal(auditResult.rawValue, false);
     assert.equal(auditResult.extendedInfo.value.length, 3);
   });
+
+  it('fails when listener hasn`t url', () => {
```
Comment (ebidel) [inline lighthouse-core/test/audits/dobetterweb/no-mutation-events-test.js]: fails when listener is missing a url property

File: lighthouse-core/test/audits/dobetterweb/uses-passive-event-listeners-test.js [hunk 3]
```diff
@@ -69,4 +69,14 @@ describe('Page uses passive events listeners where applicable', () => {
     assert.equal(auditResult.rawValue, true);
     assert.equal(auditResult.extendedInfo.value.length, 0);
   });
+
+  it('fails when listener hasn`t url', () => {
```
Comment (ebidel) [inline lighthouse-core/test/audits/dobetterweb/uses-passive-event-listeners-test.js]: fails when listener is missing a url property

File: lighthouse-core/test/audits/dobetterweb/no-datenow-test.js [hunk 4]
```diff
@@ -66,4 +66,21 @@ describe('Page does not use Date.now()', () => {
     assert.equal(auditResult.rawValue, false);
     assert.equal(auditResult.extendedInfo.value.length, 2);
   });
+
+  it('fails when usage hasn`t url', () => {
```
Comment (ebidel) [inline lighthouse-core/test/audits/dobetterweb/no-datenow-test.js]: maybe "only passes when has url property"

File: lighthouse-core/test/audits/dobetterweb/no-datenow-test.js [hunk 5]
```diff
@@ -66,4 +66,21 @@ describe('Page does not use Date.now()', () => {
...
+          {url: 'http://example.com/two', line: 10, col: 1},
+          {url: 'http://example2.com/two', line: 2, col: 22}
+        ]
+      },
+      URL: {finalUrl: URL},
+    });
+
+    assert.equal(auditResult.rawValue, false);
+    assert.ok(auditResult.extendedInfo.value[0].url === '');
+    assert.equal(auditResult.extendedInfo.value.length, 2);
```
Comment (ebidel) [inline lighthouse-core/test/audits/dobetterweb/no-datenow-test.js]: Can you check that `extendedInfo.value` has a url and is what you expect? From the first assert, it looks like `extendedInfo.value` will be the first to values in `DateNowUse.usage`?

File: lighthouse-core/test/audits/dobetterweb/no-datenow-test.js (hunk 4 above)
Comment (mixed) [inline lighthouse-core/test/audits/dobetterweb/no-datenow-test.js]: DONE!

**Pull request merged**